plugins/pluginmanagers/lazy.nix

```diff
@@ -1,222 +1,288 @@
+
+                cmd = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on command";
+
+                ft = mkNullOrOption (maybeRaw (either str (listOf str))) ''
+                  Lazy-load plugin on filetype e.g.:
+
+                  plugins.lazy.plugins = with pkgs.vimPlugins; [
+                    {
+                      pkg = neorg;
+                      # Only load plugin on "norg" filetyes
+                      ft = "norg";
+                      opts = {
+                        load."['core.defaults']" = [ ];
+                      };
+                    }
+                  ];
+                '';
+
+                keys = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
+
+                module = mkNullOrOption (enum [ false ]) ''
+                  Do not automatically load this Lua module when it's required somewhere.
+                '';
+
+                priority = mkNullOrOption number ''
+                  Only useful for start plugins i.e.
+                  `plugins.lazy.plugins.<plugin>.lazy = false` to force loading
+                  certain plugins first. Default priority is 50. It's
+                  recommended to set this to a high number for colorschemes.
+                '';
+
+                optional = defaultNullOpts.mkBool false ''
+                  Optional specs are only included in the final configuration
+                  if the corresponding plugin is also specified as a required
+                  (non-optional) plugin elsewhere. This feature is
+                  particularly helpful for Neovim distributions, allowing
+                  them to pre-configure settings for plugins that users may
+                  or may not have installed.
+                '';
+
+                opts = mkNullOrOption (maybeRaw (attrsOf anything)) ''
+                  The opts value can be one of the following:
+
+                  - A table: This table will be merged with any existing
+                  configuration settings from parent specifications.
+                  - A function that returns a table: The returned table will
+                  completely replace any existing configuration settings from
+                  parent specifications.
+                  - A function that modifies a table: This function will
+                  receive the existing configuration table as an argument and
+                  can modify it directly.
+
+                  In all cases, the resulting configuration table will be
+                  passed to the Plugin.config() function. Setting the opts
+                  value automatically implies that Plugin.config() will be
+                  called. (See: https://lazy.folke.io/spec#spec-setup)
+                '';
+              };
+
+              config.name = lib.mkIf (cfg.pkg != null) (lib.mkDefault "${lib.getName cfg.pkg}");
+              config.dir = lib.mkIf (cfg.pkg != null) (lib.mkDefault "${cfg.pkg}");
```

Comment:
```suggestion
              config = lib.mkIf (cfg.pkg != null) {
                name = lib.mkDefault "${lib.getName cfg.pkg}";
                dir = lib.mkDefault "${cfg.pkg}";
              };
```

Reply:
TBH I don't think we even need separate `pkg` and `dir` options...

Couldn't we have a `source` option that is `either path package`?

We could still have a default for name, e.g:
```nix
config.name = mkOptionDefault (
  if isDerivation config.source then
    getName config.source
  else
    baseNameOf config.source
);
```

I'm also debating `mkDefault` vs `mkOptionDefault` here. The latter is lower priority (higher priority number); its the same as a default passed directly to `mkOption`.

Reply:
That sounds like a valid reason to have it accept any string. Makes me wonder if "source" is an appropriate name though, if it can also be a "plugin name"?

If we like the idea of a type-checked option named source, we could also have a "name" option. Bearing in mind it doesn't necessarily have to correspond 1-to-1 with upstream's plugin spec, since we can do conversion when assigning to `settings.spec`.

No strong preferences here, just exploring possible options :slightly_smiling_face: